examples/nidcpower/measure-record.py

```diff
@@ -0,0 +1,196 @@
+#
+# Getting Started:
+#
+# To run this example, install "NI-DCPower Driver" on the server machine.
+# Link : https://www.ni.com/en-in/support/downloads/drivers/download.ni-dcpower.html
+#
+# For instructions on how to use protoc to generate gRPC client interfaces, see our "Creating a gRPC Client" wiki page.
+# Link: https://github.com/ni/grpc-device/wiki/Creating-a-gRPC-Client
+#
+# Refer to the NI DCPOWER gRPC Wiki for the latest C Function Reference:
+# Link: https://github.com/ni/grpc-device/wiki/NI-DCPOWER-C-Function-Reference
+#
+# Running from command line:
+#
+# Server machine's IP address, port number, and resource name can be passed as separate command line arguments.
+#   > python measure-record.py <server_address> <port_number> <resource_name>
+# If they are not passed in as command line arguments, then by default the server address will be "localhost:31763", with "SimulatedDCPower" as the resource name
+
+import grpc
+import sys
+import time
+import nidcpower_pb2 as nidcpower_types
+import nidcpower_pb2_grpc as grpc_nidcpower
+import matplotlib.pyplot as plt
+import keyword
+import numpy as np
+
+server_address = "localhost"
+server_port = "31763"
+session_name = "NI-DCPower-Session"
+
+# Resource name, channel name and options for a simulated 4147 client. Change them according to NI-DCpower model.
+resource = "SimulatedDCPower"
+options = "Simulate=1,DriverSetup=Model:4147;BoardType:PXIe"
+channels = "0"
+
+# Parameters
+record_length = 100
+voltage_level = 5.0
+
+# Read in cmd args
+if len(sys.argv) >= 2:
+    server_address = sys.argv[1]
+if len(sys.argv) >= 3:
+    server_port = sys.argv[2]
+if len(sys.argv) >= 4:
+    resource = sys.argv[3]
+    options = ""
+
+# Checks for errors. If any, throws an exception to stop the execution.
+any_error = False
+def CheckForError (vi, status) :
+    global any_error
+    if(status != 0 and not any_error):
+        any_error = True
+        ThrowOnError (vi, status)
+
+def ThrowOnError (vi, error_code):
+    error_message_request = nidcpower_types.ErrorMessageRequest(
+        vi = vi,
+        error_code = error_code
+        )
+    error_message_response = client.ErrorMessage(error_message_request)
+    raise Exception (error_message_response)
+
+# Create the communication channel for the remote host and create connections to the NI-DCPower and session services.
+channel = grpc.insecure_channel(f"{server_address}:{server_port}")
+client = grpc_nidcpower.NiDCPowerStub(channel)
+
+try :
+    # Initialise the session
+    initialize_with_independent_channels_response = client.InitializeWithIndependentChannels(nidcpower_types.InitializeWithIndependentChannelsRequest(
+        session_name = session_name,
+        resource_name = resource,
+        reset = False,
+        option_string = options
+    ))
+    vi = initialize_with_independent_channels_response.vi
+    CheckForError(vi, initialize_with_independent_channels_response.status)
+
+    # configure measure_when attribute
```

Comment:
I don't think that restating the attribute name provides much value here. It would be more helpful to use language from the help doc for the attribute: "Specify when the measure unit should acquire measurements"

https://zone.ni.com/reference/en-XX/help/370736U-01/dcpowerpropref/pnidcpower_measurewhen/

Reply:
Same for other sets below

Reply:
Done.